### ISSUE TITLE
Error in step 3 of open reference picking

### ISSUE THREAD

**Issue by mlangill** [biocore/qiime#1835]
I am getting an error during the command filter_fasta.py during step 3 of pick_open_reference_otus.py:

>
Command run was:
 filter_fasta.py -f sortmerna_otus/sortmerna_picked_otus/combined_seqs_failures.fna -s /home/mlangill/Dropbox/work/projects/andy/ucrss_sortmerna_sumaclust//step3_otus//failures_failures.txt -o /home/mlangill/Dropbox/work/projects/andy/ucrss_sortmerna_sumaclust//step3_otus//failures_failures.fasta
Command returned exit status: 2
Stdout:
Stderr
Error in filter_fasta.py: option -s: file does not exist: '/home/mlangill/Dropbox/work/projects/andy/ucrss_sortmerna_sumaclust//step3_otus//failures_failures.txt'

The file is in that directory but it is called "combined_seqs_failures_failures.txt".

My full log is here: https://www.dropbox.com/s/to0esua01bvs9if/log_20150108131428.txt?dl=0

Note that I am starting the script at step 2 with this command:
>
pick_open_reference_otus.py -i /home/mlangill/Dropbox/work/projects/andy/combined_fasta/combined_seqs.fna -o $PWD/ucrss_sortmerna_sumaclust/ -p $PWD/ucrss_smr_suma_params.txt -m sortmerna_sumaclust --step1_otu_map_fp sortmerna_otus/sortmerna_picked_otus/combined_seqs_otus.txt --step1_failures_fasta_fp sortmerna_otus/sortmerna_picked_otus/combined_seqs_failures.fna -s 0.01 -v

My params file contains this information:
>
pick_otus:sortmerna_db /home/shared/sortmerna/97_otus
pick_otus:threads 4

**Comment by mlangill** [comment]
I may have a fix for this but I won't be done testing until after the weekend. If successful I will issue a PR.

**Comment by gregcaporaso** [comment]
Thanks @mlangill, we'll keep an eye out for your PR.

Can you describe the scenario that led to your using that recovery option (i.e., restarting after step 1)? That was something that I put in place very early in the development of this workflow, but which I haven't used in years. I've been considering dropping support for that as I've been thinking that it's not useful enough to justify continuing to support it.

**Comment by mlangill** [comment]
I haven't investigated enough, but I am guessing the renaming problem is arising from me starting at step 2, otherwise I presume you would have caught this error by now. 

I do like this option, since I did step 1 as part of the closed_reference_otu_picking.py step first, then I wanted to add on the de novo otus. One thing that would be nice with this option though is if it took in the failures.txt file instead of failures.fna files since the closed pipeline doesn't create the fasta file for you. Just means I had to figure out on my own to run filter_fasta.py command before I could use the open reference pipeline. 

Would also be nice if there is a failure in the pipeline there would be an easier way to resume. I tried using the -f option with the same output directory, but it started at the first again.

**Comment by gregcaporaso** [comment]
Thanks for the details @mlangill. The use case that you're describing isn't something that we recommend doing. The preferred approach for achieving this is [outlined here](http://qiime.org/tutorials/open_reference_illumina_processing.html#filtering-an-open-reference-otu-table-to-reference-otus-only) - briefly, you build the open reference table, and then filter that to closed-reference otus only. 

Better failure recovery options are something that we really want to support, but will need to go into QIIME 2. It's not feasible to add for QIIME 1.9.0, in large part due to how cumbersome the existing workflow framework is (which I'm sure you noticed as you were working on your recent PR). This will be completely re-written for QIIME 2, so that's when we'll have the bandwidth to move it forward. 

Given all of this, and that the current failure recover options in ``pick_open_reference_otus.py`` are buggy, I'm going to remove support rather than fix the bugs, so don't worry about issuing a fix.

Thanks again for your help with testing/fixing issues in 1.9.0-rc1, and sorry we can't address these specific requests right now.

**Comment by gregcaporaso** [comment]
Note: I made some edits to the above comment after discussion with @jairideout. If you're viewing my previous comment via e-mail, be sure to look at it on GitHub for the updates.

**Comment by mlangill** [comment]
Yes, this approach would not be a normal use case. It was more that I had done closed reference otu picking already and then later decided to add on de novo otus. Since step 1 was exactly the same as what was already performed in closed reference picking I decided to use this shortcut. 

That all being said I understand that it may at this point not be worth keeping support for these options since it isn't a usual workflow. 

I will run the data from the beginning of the pipeline, which should work without any fixes.

**Comment by gregcaporaso** [comment]
closing in favor of #1841.
